gammapy/cube/fit.py

```diff
@@ -1431,6 +1464,63 @@ def cutout(self, position, width, mode="trim"):
 
         return cutout_dataset
 
+    def to_image(self, spectrum=None, name=None):
+        """Create images by summing over the energy axis.
+
+        Exposure is weighted with an assumed spectrum,
+        resulting in a weighted mean exposure image.
+
+        Currently the PSFMap and EdispMap are dropped from the
+        resulting image dataset.
+
+        Parameters
+        ----------
+        spectrum : `~gammapy.modeling.models.SpectralModel`
+            Spectral model to compute the weights.
+            Default is power-law with spectral index of 2.
+        name : str
+            Name of the new dataset.
+
+        Returns
+        -------
+        dataset : `MapDatasetOnOff`
+            Map dataset containing images.
+        """
+        dataset = super().to_image(spectrum, name)
+
+        if self.mask_safe is None:
+            mask_safe = 1
+        else:
+            mask_safe = self.mask_safe
+
+        kwargs = {}
+
+        if self.counts_off is not None:
+            counts_off = self.counts_off * mask_safe
+            kwargs["counts_off"] = counts_off.sum_over_axes(keepdims=True)
+
+        if self.acceptance is not None:
+            acceptance = self.acceptance * mask_safe
+            kwargs["acceptance"] = acceptance.sum_over_axes(keepdims=True)
+            background = self.background * mask_safe
+            background = background.sum_over_axes(keepdims=True)
+            kwargs["acceptance_off"] = (
+                kwargs["acceptance"] * kwargs["counts_off"] / background
+            )
+        kwargs["counts"] = dataset.counts
```

Comment:
For the `SpectrumDatasetOnOff` I have introduced `.from_spectrum_dataset(dataset, acceptance, acceptance_off)` at some point. This might be helpful here as (and probably in the ring background estimators as well...)

Reply:
Yes absolutely. I propose to add this in a future PR and adapt there.